core/trino-main/src/main/java/io/trino/sql/planner/plan/AggregationNode.java

```diff
@@ -65,6 +65,22 @@ public static AggregationNode simpleSingleAggregation(
         return new AggregationNode(id, source, aggregations, groupingSets, ImmutableList.of(), SINGLE, Optional.empty(), Optional.empty());
     }
 
+    public static AggregationNode restoreDistinctAggregation(
```

Comment:
What does this method do actually?

This method is very specific to some optimization rules. I would keep it there

Reply:
I think keeping this method is useful. It gives semantics to the complex constructor invocation that is copied in multiple cases.
It may be specific to the correlated aggregation optimization rules so I moved it to `AggregationDecorrelation`